### PR TITLE
fix(studio): harden cluster, ACL, client, and polling behavior

### PR DESCRIPTION
**Pull request by yyqdbngt**
## Summary

This consolidates the remaining small Studio changes requested in the review queue into one submission with cohesive commits:

- isolate nested cluster repository snapshots and copied configurations
- reject duplicate instance and catalog provider registrations
- serialize visible-page polling to prevent overlapping requests
- support IPv6 literals, CIDR ranges, and wildcards in ACL matching
- scope mock client inventories by the NameServer address required by the current real endpoint
- mask credentials on Unicode code-point boundaries and handle malformed surrogates safely

The branch is based on the latest `rocketmq-studio`. During consolidation, the cluster snapshot change was adapted to the current `gmtModified` model, and mock client isolation was updated for the current NameServer-based client API introduced by the recent instance refactor.

The system-topic change from #2245 was absorbed upstream by #2365 during the latest rebase and is therefore no longer duplicated in this PR's diff. This PR otherwise supersedes #2217, #2218, #2241, #2247, #2251, and #2253.

## Validation

- `mvn -q -Dtest=ClusterRepositoryImplTest,InstanceProviderRegistryTest,SystemTopicFilterTest,IpRangeMatcherTest,CredentialUtilsTest test`
- `npm exec vitest -- run src/hooks/useVisiblePolling.test.tsx src/services/connectionsService.test.ts` (9 tests passed)
- `npm run build`
- `npm run lint` (0 errors; one pre-existing exhaustive-deps warning in `src/pages/instance/topic.tsx`)
- Prettier check on all changed frontend files
- `git diff --check`

### CODE REVIEW

**Review by RockteMQ-AI — COMMENTED** [review]
### Summary
Review completed based on diff analysis. No structured findings extracted from engine output.

### Findings
No specific line-level findings. Overall assessment provided in summary.

---
*Automated review by RockteMQ-AI*

**Comment by RockteMQ-AI** [comment]
⚠️ **Merge conflict detected**

This PR has conflicts with the base branch and cannot be merged. Please rebase or merge the base branch into your branch and resolve the conflicts:

```bash
git fetch origin
git checkout codex/wang-studio-consolidated
git rebase origin/develop
# resolve conflicts, then:
git push --force-with-lease
```

This is a one-time reminder. Feel free to @mention me for a re-review after conflicts are resolved.

---
*Automated notification by github-manager-bot*

**Comment by yyqdbngt** [comment]
Rebased onto the latest `rocketmq-studio` and resolved the conflict by dropping the now-upstream system-topic commit (#2365). The remaining six commits were revalidated locally; backend targeted tests, 9 frontend tests, the production build, lint, Prettier, and `git diff --check` all pass. @RockteMQ-AI please re-review when ready.

**Review by RockteMQ-AI — APPROVED** [review]
### Summary
This PR consolidates multiple Studio improvements into one cohesive submission: deep-copy isolation for nested cluster snapshots, Unicode-safe credential masking, IPv6/CIDR ACL matching, duplicate registration prevention, serialized polling, and mock client scoping by NameServer address.

The changes are well-structured, each addressing a specific correctness or security concern. Test coverage is thorough — both Java and frontend tests validate the new behavior including edge cases (malformed surrogates, supplementary code points, overlapping poll requests, empty address tables).

**Verdict: LGTM.** Clean, defensive code with good test discipline.

---
*Automated review by github-manager-bot*

File: server/src/main/java/org/apache/rocketmq/studio/instance/acl/IpRangeMatcher.java [hunk 1]
```diff
@@ -105,23 +107,37 @@ public static boolean isInRange(String ip, String cidrOrIp) {
...
+            String[] parts = value.split("\\.");
+            byte[] bytes = new byte[4];
+            for (int i = 0; i < bytes.length; i++) {
+                bytes[i] = (byte) Integer.parseInt(parts[i]);
+            }
+            return bytes;
+        }
+        if (value == null || !value.contains(":")
+                || !IPV6_LITERAL_CHARACTERS.matcher(value).matches()) {
+            return null;
```
Comment (RockteMQ-AI) [inline server/src/main/java/org/apache/rocketmq/studio/instance/acl/IpRangeMatcher.java:126]: **[Info]** `InetAddress.getByName(value)` is guarded by the `IPV6_LITERAL_CHARACTERS` regex which prevents DNS resolution — good. One minor note: Java's `InetAddress.getByName` for IPv6 literals can internally normalize mapped addresses (e.g. `::ffff:192.168.1.1` returns a 4-byte `Inet4Address`). The `instanceof Inet6Address` check on line 127 correctly rejects those, so IPv4-mapped IPv6 entries silently return `null`. This is safe but worth a comment for future maintainers.

**Review by RockteMQ-AI — APPROVED** [review]
### Summary
Re-reviewed after rebase. The PR consolidates six Studio hardening commits covering deep defensive copies, IPv6 ACL support, Unicode-safe credential masking, duplicate provider detection, poll overlap prevention, and NameServer-scoped mock data. All changes are well-structured with comprehensive test coverage.

### Rebase Verification
- Dropped system-topic commit (#2365) — now upstream ✓
- Remaining six commits revalidated ✓
- No new issues introduced by the rebase

### Findings
- **[Info]** IpRangeMatcher.java:126 — IPv4-mapped IPv6 addresses silently return null from `parseAddressLiteral`; safe but could use a comment
- All other changes look correct and well-tested

### Verdict
**APPROVED** — clean rebase, high-quality consolidation of Studio hardening fixes.

---
*Automated review by RockteMQ-AI*
